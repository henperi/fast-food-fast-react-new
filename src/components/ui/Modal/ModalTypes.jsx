import React from 'react';

export const AddToCartModal = (props) => {
  const {
    title, price, description, handleModal,
  } = props;
  console.log('insideAddToCart', props);
  return (
    <div>
      <h2 className="text-center">Add this food item to existing Cart</h2>
      <div className="content-div">
        <div className="item-title">
          {title}
-
          <span className="badge price">
            &#8358;
            {price}
          </span>
        </div>
        <div className="item-description">
          <p>{description}</p>
        </div>
      </div>

      <button type="button" className="btn btn-blue btn-block btn-rounded btn-bg">
        Add To Cart
      </button>

      <button
        onClick={handleModal}
        type="button"
        className="close-button btn btn-primary btn-block btn-rounded btn-bg"
      >
        Cancel
      </button>
    </div>
  );
};

export const RemoveFromCartModal = (props) => {
  const {
    title, price, description, handleModal,
  } = props;

  return (
    <div>
      <h2 className="text-center">Remove this food item from existing Cart</h2>
      <div className="content-div">
        <div className="item-title">
          {title}
-
          <span className="badge price">
            &#8358;
            {price}
          </span>
        </div>
        <div className="item-description">
          <p>{description}</p>
        </div>
      </div>

      <button type="button" className="btn btn-red btn-block btn-rounded btn-bg">
        Remove From Cart
      </button>

      <button
        onClick={handleModal}
        type="button"
        className="close-button btn btn-primary btn-block btn-rounded btn-bg"
      >
        Cancel
      </button>
    </div>
  );
};

export const SignUpModal = (props) => {
  const { handleModal } = props;

  return (
    <div>
      <h2 className="text-center">Signup Now</h2>
      <div className="content-div">
        <div className="item-description">
          <input />
        </div>
      </div>
      <span>
        <button type="button" className="btn btn-green btn-blockX btn-rounded btn-bg">
          Sign Up
        </button>

        <button
          onClick={handleModal}
          type="button"
          className="close-button btn btn-default btn-blockX btn-rounded btn-bg"
        >
          Cancel
        </button>
      </span>
    </div>
  );
};
