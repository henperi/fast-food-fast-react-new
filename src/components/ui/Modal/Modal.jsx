import React from 'react';
import { AddToCartModal, RemoveFromCartModal, SignUpModal } from './ModalTypes';

const modal = (props) => {
  const {
    id, isOpen, type, handleModal,
  } = props;
  return (
    <div className={`modal ${isOpen && 'show-modal'}`} id={id}>
      <div className="modal-content">
        <div className="text-center">
          <button
            type="button"
            onClick={handleModal}
            className="close-button btn btn-sm push-right fs-20"
          >
            x
          </button>
          {type === 'addToCart' && <AddToCartModal {...props} />}
          {type === 'removeFromCart' && <RemoveFromCartModal {...props} />}
          {type === 'signUpModal' && <SignUpModal {...props} />}
        </div>
      </div>
    </div>
  );
};

export default modal;
