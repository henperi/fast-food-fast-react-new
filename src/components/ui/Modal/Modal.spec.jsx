import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

test('should render Modal correctly', () => {
  const wrapper = shallow(<Modal />);
  expect(wrapper).toMatchSnapshot();
});
