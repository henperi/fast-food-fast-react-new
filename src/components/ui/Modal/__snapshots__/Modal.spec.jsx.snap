// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Modal correctly 1`] = `
<div
  className="modal undefined"
>
  <div
    className="modal-content"
  >
    <div
      className="text-center"
    >
      <button
        className="close-button btn btn-sm push-right fs-20"
        type="button"
      >
        x
      </button>
    </div>
  </div>
</div>
`;
